test/e2e/validation_test.go

```diff
@@ -19,78 +19,94 @@ limitations under the License.
 package e2e_test
 
 import (
+	"context"
+	"fmt"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	capiv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/framework"
 	"sigs.k8s.io/cluster-api/util"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var _ = Describe("Cluster API Validation", func() {
-	It("verifies validation is setup correctly and works", func() {
-		bootstrapData := "some valid data"
-		name := "capi-test-machine-" + util.RandomString(6)
-		machine := &capiv1.Machine{
-			ObjectMeta: v1.ObjectMeta{
-				Name:      name,
-				Namespace: capiTestNamespace,
-			},
-			Spec: capiv1.MachineSpec{
-				ClusterName: name,
-				Bootstrap: capiv1.Bootstrap{
-					Data: &bootstrapData,
+
+	var capiTestNamespace string
+
+	BeforeEach(func() {
+		// Create namespace before each context for better separation of
+		// tests.
+		capiTestNamespace = "capi-validation-test-" + util.RandomString(4)
+		framework.EnsureNamespace(context.Background(), kindClient, capiTestNamespace)
+	})
+
+	Context("for Machine", func() {
```

Comment:
This should create a namespace for each context block. That way it is granular to specific validation hook types.